src/python/TaskWorker/Actions/RetryJob.py

```diff
@@ -338,7 +338,10 @@ def execute_internal(self, reqname, job_return_code, crab_retry, job_id, dag_job
         self.job_id          = job_id
         self.dag_jobid       = dag_jobid
 
-        self.get_job_ad()
+        if not use_root_job_ad:
+            self.get_job_ad()
```

Comment:
An important part of get_job_ad() is that it puts all execution attempts into self.ads -- not just the latest one!  This is critical for check_cpu_report.

I think you need to append the latest job ad to finished_jobs/$DAG_NODE_ID.ads, then have get_job_ad parse that file instead of doing its condor_q.